### ISSUE TITLE
Daedalus Spending Password issue with Trezor

### ISSUE THREAD

**Issue by oranheim** [input-output-hk/daedalus#2493]
I tried the latest Daedalus v4.0.3 feature for Trezor Hidden Wallets and lost my ADA? Not good, making me very nervous..

Issue: unable to access 2 ADA -- spending password NOT working

Reproduce issue:

1) Create hidden wallet with a 4 char 13th mnemonic
2) I created a Daedalus wallet and importing public key using #1
3) Transferred 2 ADA to the hidden wallet
4) Transaction was successful

--so far so good--

Problems occurs due to below:

5) Deleted the Hidden Daedalus Wallet
6) Re-created a Daedalus wallet, ref #2
7) I am asked for setting a Spending Password (not asked about that earlier in any procedure)
8) Unable to do #7 and henceforth unable to restore my funds

Are my ADA lost? Can I trust Daedalus?

**Comment by nikolaglumac** [comment]
Something must have gone wrong with the earlier attempts to pair your Trezor wallet.
Do you have any other wallets in Daedalus or is it just your Trezor wallet?

**Comment by nikolaglumac** [comment]
Can you please send your logs to my email address: nikola.glumac@iohk.io or submit a support ticket here: https://iohk.zendesk.com/hc/en-us/requests/new and make sure to include the logs?

**Comment by oranheim** [comment]
> Something must have gone wrong with the earlier attempts to pair your Trezor wallet.
> Do you have any other wallets in Daedalus or is it just your Trezor wallet?

Nope, only, Trezor Model T.

**Comment by oranheim** [comment]
> Can you please send your logs to my email address: [nikola.glumac@iohk.io](mailto:nikola.glumac@iohk.io) or submit a support ticket here: https://iohk.zendesk.com/hc/en-us/requests/new and make sure to include the logs?

Sent.

**Comment by nikolaglumac** [comment]
Thanks @oranheim! I have received the logs.
@tomislavhoracek, our lead hardware wallet developer, will check it out next week once he returns from his vacation.
Thanks for the patience!

**Comment by oranheim** [comment]
Great! :)

**Comment by tomislavhoracek** [comment]
@oranheim I'm investigating the issue you found. Can you tell me what message is shown on the Trezor device for **case 7.**? Maybe that is connected to the Trezor-Connect issue we also faced?
https://github.com/trezor/connect/issues/783

Can tell me what firmware version you have on the device?

**Comment by oranheim** [comment]
> @oranheim I'm investigating the issue you found. When you were restoring a wallet in case 1-2 you were asked to enter a passphrase on the device, right?
> Can you tell me what firmware version you have on the device?

My Trezor Model T firmware is: 2.3.6

> Also one more question:
> in step 6: _Re-created a Daedalus wallet, ref #2_
> You restored the hidden wallet (one with a passphrase) or the original one ( without a passphrase ) after you deleted the hidden wallet?

It was the one with a passphrase (the hidden wallet).

I'll try your to remove the State directory and try to reinstall a fresh copy of Daedalus.

**Comment by oranheim** [comment]
I was able to restore my "hidden wallet #1". However, I am unable to restore the "standard wallet".

Does this mean that Daedalus only supports one paired wallet at a time?

I also think that found the bug that introduces "set spending password" issue.

How to reproduce:

1) Set up Trezor Suite with a Hidden Wallet #1
2) Pair your Model T device and enter your 13th mnemonic key (Please note: PIN is used for Standard Wallet)
3) Blockchain sync takes some time
4) The wallet is successfully restored

5) Pair your Model T device and skip entering a passphrase (13th mnemonic)
6) The public key shows up on Model T and accept chose accept
7) The wallet shows up with the name=Trezor Wallet, with no pairing symbol attached.
8) It is now asking for the Spending Password that cannot be set <= BUG

I would expect Daedalus to support both Standard and Hidden Wallet at the same time. The Standard requires a PIN to be entered. The Hidden a 13th mnemonic passphrase.

**Comment by oranheim** [comment]
I found a way to make this work :)

1) Set up Trezor in Standard Wallet mode (using pin)
2) Open Daedalus and Pair (follow instructions)
3) Set up Trezor in Hidden Wallet mode (with Passphrase)
4) Open Daedalus and Pair (follow instructions)

When you want to send money from Standard Wallet, you need to disable the Trezor Passphrase; and re-enable it when you want to operate on the Hidden Wallet.

**Comment by tomislavhoracek** [comment]
@oranheim We are glad to hear you successfully restored your hidden wallet.
Thanks for the details (reproduction steps). We will investigate the issue. Daedalus supports both Standard and Hidden wallets at the same time without disabling/enabling passphrase.

**Comment by nikolaglumac** [comment]
@tomislavhoracek @ManusMcCole  please try to reproduce the issue @oranheim reported 🙏 
It should be possible to restore a no-passphrase wallet even without disabling passphrase on the device.

**Comment by tomislavhoracek** [comment]
@nikolaglumac I agree. I'm on it!

**Comment by oranheim** [comment]
Thanks a lot guys! I really appreciate your help and attention. Awesome 🥇

**Comment by nikolaglumac** [comment]
We will sort it out together - I am sure of that 🙌   😄

**Comment by ManusMcCole** [comment]
I was able to reproduce the issue above in 1 out of 6 attempts(I cleaned state and reinstalled Daedalus between each attempt) by

1. Pairing a standard wallet(No passphrase)
2.Unplugging device
3.Plugging in device.
4.Pair a new wallet with passphrase

Daedalus then wrongly asks for a spending password for the passphrase protected as also outlined above. The issue only occurred 1 out of 6 times with normal functionality occurring the rest of the time

**Comment by nikolaglumac** [comment]
@tomislavhoracek please take it from here 🙏

**Comment by ManusMcCole** [comment]
Hi @tomislavhoracek. When the issue occurs on windows the error is different. Instead of asking for a spending password the user gets an error occured message when trying to pair the wallet with this [message](https://drive.google.com/file/d/1sFpSRr2uMd-prqMSaIU-a4fvGOQT9SC_/view?usp=sharing) in the console(unexpected error). I hope this helps you to figure out the cause

**Comment by nikolaglumac** [comment]
The [PR 2520](https://github.com/input-output-hk/daedalus/pull/2520) with the fix has been merged and it will be released in one of the upcoming Daedalus releases.
Thanks for the fix @tomislavhoracek and @ManusMcCole, and thanks for reporting this and helping us debug it @oranheim 🙏

**Comment by nikolaglumac** [comment]
Hi all,

the fix for this issue has been released on Daedalus Flight within the 4.1.0-FC1 version: https://github.com/input-output-hk/daedalus/releases/tag/4.1.0-FC1
You can use the automatic update within existing  to get the new version or download it on the official Daedalus website: https://daedaluswallet.io/en/flight/

_PS: We will release the same fix on Daedalus Mainnet tomorrow around 16:00 UTC._

Thanks for the patience!

**Comment by nikolaglumac** [comment]
Daedalus Mainnet 4.1.0 has been released and includes this fix.
Thanks for the patience!

**Comment by dave8000** [comment]
I have Daedalus Mainnet 4.1.0 installed and I have a similar issue, so the 4.1.0 may not complete fix the issue.  I can see both wallets but the standard wallet, (without passphrase), which I set up second, came back with, "Your wallet is not protected with a spending password."  It then requests aa password, which does not function.    The trezor standard wallet set up is stuck at this point.  I can see ADA amount grayed out so it has the right public keys.  I can see the standard wallet delgated to the correct pool with correct amount.  Wallet name DOES NOT have the hardware key emblem next to it.  Trezor Passphrase wallet, loaded correctly, no pass phrase requested, Hardware key emblem is shown on that wallet name.  

To recap, standard wallet is stuck at the false step of asking me to set a password.

**Comment by dave8000** [comment]
All functional access to the wallet is blocked since it is stuck at the point of requesting me to set a spending password.  Hence I cannot delete this standard wallet.  If I try to install again I get the message, "Wallet you are trying to restore already exists."     I really don't want to have to reinstall Daedalus from scratch on this machine as there are additional wallets to all get resynced.      It may come down that if there is not another solution.

**Comment by dave8000** [comment]
On second computer, Daedalus Mainnet, both standard and Password Trezor wallets installed correctly with no issues.  Installed standard wallet first and Password wallet second on this second computer.   First computer with Daedalus 4.1.0 still stuck asking for spending password for standard wallet.

**Comment by nikolaglumac** [comment]
@dave8000 I guess your local state is broken. @tomislavhoracek please help Dave here 🙏

**Comment by tomislavhoracek** [comment]
@dave8000 yeah, as Nikola said, your local state is broken. The only solution is to clean the state directory.
Here are instructions https://iohk.zendesk.com/hc/en-us/articles/360039511853-Delete-the-contents-of-the-state-directory
To prevent syncing from scratch you can leave `chain` folder in the state directory and just remove other files and folders.

P.S. Ensure you have recovery phrases for your wallets.
